pkg/reconciler/knativeeventing/knativeeventing.go

```diff
@@ -107,11 +111,11 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ke *eventingv1alpha1.Kna
 	return nil
 }
 
-// transform re-assigns the passed manifest to one with common and
+// transform mutates the passed manifest to one with common and
 // platform transforms, plus any extras passed in
 func (r *Reconciler) transform(ctx context.Context, manifest *mf.Manifest, instance *eventingv1alpha1.KnativeEventing) error {
 	logger := logging.FromContext(ctx)
-	return common.Transform(ctx, manifest, instance, r.kubeClientSet,
+	return common.Transform(common.WithPlatforms(ctx, r.platform), manifest, instance, r.kubeClientSet,
```

Comment:
If we add this only here anyway, shall we just pass it? That at least makes it typesafe and we pass a ton of parameters already anyway.

Reply:
Good point. I went a step further, making platform more abstract until we know better how it's going to be used.